### DISCUSSION TITLE
Toffy tiling top left corner not accurate

### DISCUSSION THREAD

**Discussion by akandrew332** [angelolab/toffy#273] [Q&A]
I've been trying tiling using Toffy.  However, I can't seem to get the top left corner for the first panel to reflect the area that I picked in MIBIcontrol as the top left corner.  I have tried recalibrating stage to optical coordinates and that hasn't worked.  I've attached an image of the preview tiling area generated in Toffy.  The red dot indicates the target coordinates selected in MIBIcontrol.  Does the target area selected in MIBI control translate as the centroid of the top left corner FOV in Toffy?  If that is the case, then the two are misregistered by about 600 microns in x and 200 microns in y (FOV is 400).  I'm likely doing something wrong, and ideas as to what that might be?
![test1afterrecalibrationANDafterTARGETselection](https://user-images.githubusercontent.com/19846608/197047322-fe616d66-f0c9-4f9d-b6fd-6e3293f4d704.JPG)

**Comment by akandrew332** [comment]
I have noticed that in MIBIcontrol the target area selected and the target area registered after left click never match up.  Not sure if this is contributing to the issue.  I tried this several times, and the end result is identical (I was pretty careful not to move the mouse while clicking on it)
Before mouse click:
![20221020_141816](https://user-images.githubusercontent.com/19846608/197051550-bb6f7110-9af1-4551-a295-5e8c382f08cf.jpg)

After mouse click:
![20221020_141842](https://user-images.githubusercontent.com/19846608/197051559-70f390e6-0f02-411f-a8dc-6cc35af9dffa.jpg)

**Reply by ngreenwald** [reply]
Hey @akandrew332, it's possible that this is an issue with MIBI control. Can you check if the coordinates of the region you're selecting are correct? i.e. is it the one you're clicking on that actually gets registered, or the one where end it ups? You should be able to navigate to the stage coordinates that are saved to confirm.

**Reply by akandrew332** [reply]
Hi, the coordinates you get are the ones where it ends up.  So to center onto a fiducial, you need to click one pixel to the right of center.  Unfortunately, this isn't enough of a difference compared to what I'm seeing.  I've had a look back at the optical to SED image registration, and it has strayed quite a bit.  I didn't originally think this was the issue since it had just been realigned.  So something has changed...

**Reply by ngreenwald** [reply]
Got it, I would reach out to ionpath in that case to get the double cursor issue resolved. Once the MIBIControl coregistration is updated, the toffy coregistration will also need to be updated in order to stay current.

**Reply by akandrew332** [reply]
Sounds good.  Thanks for your help!

**Reply by akandrew332** [reply]
On the same topic, if I've generated a tiled_json, when MIBIcontrol's co-registration was poor, then had the co-registration corrected by Ionpath, and subsequently re-registered optical-stage in Toffy, can I use the same tiled_json, or do I need to generate another tiled_json.  I'm guessing that since the json uses stage coordinates the previous tiled_json should still work.  Am I thinking about that the right way?

**Reply by ngreenwald** [reply]
I'm not sure, it depends on what changes when ionpath changes their coregistration. I don't know which set of values get updated. Sorry!

**Reply by akandrew332** [reply]
No worries. I gave it a go, and will update once I have the result.
